rust/src/protocol/mod.rs

```diff
@@ -640,6 +640,111 @@ pub struct Protocol {
     /// table.
     pub min_writer_version: i32,
+    /// Table features are missing from older versions
+    /// The table features this reader supports
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub reader_features: Option<HashSet<ReaderFeatures>>,
+    /// Table features are missing from older versions
+    /// The table features this writer supports
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub writer_features: Option<HashSet<WriterFeatures>>,
+}
+
+/// Features table readers can support as well as let users know
+/// what is supported
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum ReaderFeatures {
+    /// Mapping of one column to another
+    #[serde(alias = "columnMapping")]
+    COLUMN_MAPPING,
+    /// Deletion vectors for merge, update, delete
+    #[serde(alias = "deletionVectors")]
+    DELETION_VECTORS,
+    /// timestamps without timezone support
+    #[serde(alias = "timestampNtz")]
+    TIMESTAMP_WITHOUT_TIMEZONE,
+    /// version 2 of checkpointing
+    #[serde(alias = "v2Checkpoint")]
+    V2_CHECKPOINT,
+}
+
+#[allow(clippy::from_over_into)]
+impl Into<usize> for ReaderFeatures {
+    fn into(self) -> usize {
+        match self {
+            ReaderFeatures::COLUMN_MAPPING => 2,
+            ReaderFeatures::DELETION_VECTORS
+            | ReaderFeatures::TIMESTAMP_WITHOUT_TIMEZONE
+            | ReaderFeatures::V2_CHECKPOINT => 3,
+        }
+    }
+}
+/// Features table writers can support as well as let users know
+/// what is supported
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum WriterFeatures {
+    /// Append Only Tables
+    #[serde(alias = "appendOnly")]
+    APPEND_ONLY,
+    /// Table invariants
+    #[serde(alias = "invariants")]
+    INVARIANTS,
+    /// Check constraints on columns
+    #[serde(alias = "checkConstraints")]
+    CHECK_CONSTRAINTS,
+    /// CDF on a table
+    #[serde(alias = "changeDataFeed")]
+    CHANGE_DATA_FEED,
+    /// Columns with generated values
+    #[serde(alias = "generatedColumns")]
+    GENERATED_COLUMNS,
+    /// Mapping of one column to another
+    #[serde(alias = "columnMapping")]
+    COLUMN_MAPPING,
+    /// ID Columns
+    #[serde(alias = "identityColumns")]
+    IDENTITY_COLUMNS,
+    /// Deletion vectors for merge, update, delete
+    #[serde(alias = "deletionVectors")]
+    DELETION_VECTORS,
+    /// Row tracking on tables
+    #[serde(alias = "rowTracking")]
+    ROW_TRACKING,
+    /// timestamps without timezone support
+    #[serde(alias = "timestampNtz")]
+    TIMESTAMP_WITHOUT_TIMEZONE,
+    /// domain specific metadata
+    #[serde(alias = "domainMetadata")]
+    DOMAIN_METADATA,
+    /// version 2 of checkpointing
+    #[serde(alias = "v2Checkpoint")]
+    V2_CHECKPOINT,
+    /// Iceberg compatability support
+    #[serde(alias = "icebergCompatV1")]
+    ICEBERG_COMPAT_V1,
+    /// liquid clustering support
+    #[serde(alias = "liquid")]
+    LIQUID,
```

Comment:
Is there a `liquid` table feature? I don't see it in the [protocol](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#valid-feature-names-in-table-features).

Reply:
I only added it because I saw a test that read a liquid table and that table had liquid in the table features.